.github/workflows/e2e_on_pull.yaml

```diff
@@ -59,6 +59,9 @@ jobs:
     steps:
       - name: Checkout Code
         uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
I might have misunderstood something, but I think we had a similar thing in kata, where we pulled the `pull_request.head.sha` and had problems where if the PR wasn't rebased then you get issues, so contributors need to rebase to get accurate CI. What we ended up doing (shout-out to Jeremi for the suggestion), was switch this to `github.event.pull_request.merge_commit_sha` (some detail in the REST API [doc](https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#get-a-pull-request) and the Kata [PR](https://github.com/kata-containers/kata-containers/pull/7814#pullrequestreview-1606407631). So maybe we should consider that here as well?

Reply:
Fabiano has just posted a potential issue with this approach on slack: https://katacontainers.slack.com/archives/C879ACQ00/p1694089545591799 so it's worth understanding that before making the change

Reply:
Cool that sounds fine to me